src/main/java/frc/robot/KrakenSwerveModule.java

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot;
+
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.CANSparkBase.IdleMode;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import frc.robot.constants.KrakenSwerveConstants;
+import frc.robot.constants.SwerveConstants;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+
+/** Add your docs here. */
+public class KrakenSwerveModule {
+
+     // Parameters
+    private final int mRotID;
+    private final int mTransID;
+    private final PIDController mRotPID;
+
+    // Motors
+    private final CANSparkMax mRotMotor;
+    private final TalonFX mTransMotor;
+    
+    // Encoders
+    private final CANcoder mRotEncoder;
+    private final RelativeEncoder mRotRelativeEncoder;
+
+    // Public Debug Values
+    private double mPIDOutput = 0.0;
+    private double mDesiredRadians = 0.0;
+
+    /**
+     * Creates a new Swerve Module with a Neo for Rotation and a Kraken for Translation
+     * @param rotID Id of the Rotation Motor
+     * @param transID Id of the Translation Motor
+     * @param rotEncoderID Id of the Absolute Encoder (CANCoder)
+     * @param rotInverse Is the rotation motor inversed
+     * @param transInverse Is the translation motor inversed
+     * @param rotPID the PIDController for rotation
+     */
+    public KrakenSwerveModule(int rotID, int transID, int rotEncoderID, boolean rotInverse, boolean transInverse, PIDController rotPID) {
+        mRotID = rotID;
+        mTransID = transID;
+        mRotPID = rotPID;
+        mRotPID.enableContinuousInput(-Math.PI, Math.PI);
+
+        mRotMotor = new CANSparkMax(mRotID, MotorType.kBrushless);
+        mTransMotor = new TalonFX(mTransID);
+        
+        mRotEncoder = new CANcoder(rotEncoderID, "SWERVE_ENC");
+        mRotRelativeEncoder = mRotMotor.getEncoder();
+        mRotRelativeEncoder.setPosition(0);
+        mTransMotor.setPosition(0);
+
+        mRotMotor.setInverted(rotInverse);
+        mTransMotor.setInverted(transInverse);
```

Comment:
May not matter much, but you may want to setInverted before you setPosition - one is configuration, and the other one is command

General rule is configure first, command later.